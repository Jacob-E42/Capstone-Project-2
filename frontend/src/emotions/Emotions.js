import React from "react";
import "chart.js/auto";
import { Doughnut } from "react-chartjs-2";
import { Typography, Box } from "@mui/material";

const Emotions = ({ emotions }) => {
	if (!emotions || !emotions.joy || !emotions.sadness || !emotions.anger || !emotions.fear || !emotions.disgust)
		return null;
	// console.debug("Emotions", emotions);
	// const totalEmotions = emotions.joy + emotions.sadness + emotions.anger + emotions.fear + emotions.disgust;

	const data = {
		labels: ["Joy", "Sadness", "Anger", "Fear", "Disgust"],
		datasets: [
			{
				label: "Emotions",
				data: [emotions.joy, emotions.sadness, emotions.anger, emotions.fear, emotions.disgust],
				backgroundColor: ["#FFD700", "#0000ff", "#ed2727", "#fb9107", "#2ebc8f"]
			}
		]
	};
	const options = {
		maintainAspectRatio: true,
		responsive: true,
		redraw: false
	};

	return (
		<Box sx={{ my: 2 }}>
			<Typography
				variant="h6"
				gutterBottom>
				Emotions
			</Typography>
			{emotions && (
				<Doughnut
					data={data}
					options={options}
					datasetIdKey="totalEmotions"
					width={400}
					height={400}
				/>
			)}
		</Box>
	);
};

export default Emotions;
