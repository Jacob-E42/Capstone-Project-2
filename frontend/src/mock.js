import React, { useCallback, useState } from "react";
import UserContext from "./context_providers/UserContext";
import AlertContext from "./context_providers/AlertContext";
import Request from "./api";
import ApiRequest from "./api";
import ApiContext from "./context_providers/ApiContext";
import JournalContext from "./context_providers/JournalContext";

const demoUser = {
	id: 11,
	firstName: "testfirst",
	lastName: "testlast",
	email: "test@test.net",
	interests: ["baking", "jetskiing"]
};

const demoJournal = {
	id: 42,
	userId: 11,
	title: "Baking adventure",
	entryText:
		"Yesterday I tried baking for the first time like I've always wanted to. It was a complete disaster, but at least it was fun.",
	entryDate: "2023-07-24",
	emotions: null
};

const UserProvider = ({ children }) => {
	return <UserContext.Provider value={{ user: demoUser }}>{children}</UserContext.Provider>;
};

const AnonUserProvider = ({ children }) => {
	const [currentUser, setCurrentUser] = useState(null);

	const login = useCallback(
		async formData => {
			try {
				const request = new Request();
				const token = await request.login(formData);
				if (token) {
					setCurrentUser(demoUser);
					return { success: true };
				} else throw new Error("Log in token in missing");
			} catch (errors) {
				console.error("login failed", errors);
				return { success: false, errors };
			}
		},
		[setCurrentUser]
	);

	const signup = useCallback(
		async formData => {
			try {
				const request = new Request();
				const token = await request.signup(formData);
				console.log(token);
				if (token) setCurrentUser(demoUser);
				else throw new Error("missing sign in token");
				return { success: true };
			} catch (errors) {
				console.error("signup failed", errors);
				return { success: false, errors };
			}
		},
		[setCurrentUser]
	);

	return <UserContext.Provider value={{ currentUser, login, signup }}>{children}</UserContext.Provider>;
};

const AlertProvider = ({ children }) => {
	const [msg, setMsg] = useState("");
	const [color, setColor] = useState("primary");

	return <AlertContext.Provider value={{ msg, setMsg, color, setColor }}>{children}</AlertContext.Provider>;
};

const ApiProvider = ({ children }) => {
	const [request, setRequest] = useState(new ApiRequest());
	request.setToken(
		`eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTEsImVtYWlsIjoidGVzdEB0ZXN0Lm5ldCIsImlhdCI6MTY5MjI5MDkxN30.8e12Cw-TMts0FmZALH61ch03kUMlW5QoCi60dlfhyWY`
	);

	return <ApiContext.Provider value={{ api: request }}>{children}</ApiContext.Provider>;
};

const JournalProvider = ({ children }) => {
	return <JournalContext.Provider value={{ journal: demoJournal }}>{children}</JournalContext.Provider>;
};

export { UserProvider, AnonUserProvider, AlertProvider, ApiProvider, JournalProvider };
