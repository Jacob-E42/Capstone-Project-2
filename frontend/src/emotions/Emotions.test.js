import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { AlertProvider, ApiProvider, UserProvider } from "../mock";
import Emotions from "./Emotions";

test("Emotions renders without crashing", () => {
	render(
		<MemoryRouter>
			<UserProvider>
				<ApiProvider>
					<AlertProvider>
						<Emotions />
					</AlertProvider>
				</ApiProvider>
			</UserProvider>
		</MemoryRouter>
	);
});

test("Emotions matches snapshot", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<UserProvider>
				<ApiProvider>
					<AlertProvider>
						<Emotions emotions={"Well Done! This is an example Emotions."} />
					</AlertProvider>
				</ApiProvider>
			</UserProvider>
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

test("Emotions matches snapshot when rendered without props", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<UserProvider>
				<ApiProvider>
					<AlertProvider>
						<Emotions />
					</AlertProvider>
				</ApiProvider>
			</UserProvider>
		</MemoryRouter>
	);

	expect(asFragment()).toMatchSnapshot();
});
