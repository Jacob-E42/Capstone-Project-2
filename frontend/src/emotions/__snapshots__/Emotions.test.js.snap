// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emotions matches snapshot 1`] = `<DocumentFragment />`;

exports[`Emotions matches snapshot when rendered without props 1`] = `<DocumentFragment />`;
